app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -19,11 +21,22 @@ const getTestMiddlewares = () => {
 export default function createJsonRpcClient({ rpcUrl, chainId }) {
   const fetchMiddleware = createFetchMiddleware({ rpcUrl });
   const blockProvider = providerFromMiddleware(fetchMiddleware);
+
+  // create subscription polyfill middleware
   const blockTracker = new PollingBlockTracker({
     ...blockTrackerOpts,
     provider: blockProvider,
   });
 
+  const filterMiddleware = createFilterMiddleware({
+    blockProvider,
+    blockTracker,
+  });
+  const subscriptionManager = createSubscriptionManager({
+    blockProvider,
```

Comment:
The provider parameters have the wrong names here:

```suggestion
    provider: blockProvider,
    blockTracker,
  });
  const subscriptionManager = createSubscriptionManager({
    provider: blockProvider,
```

Reply:
Corrected here: [469bbed](https://github.com/MetaMask/metamask-extension/pull/16693/commits/469bbed580be40d466cc76a97d7910f727eea5cb)